include/rocksdb/convenience.h

```diff
@@ -46,9 +46,15 @@ struct ConfigOptions {
   // When true, any unused options will be ignored and OK will be returned
   bool ignore_unknown_options = false;
 
+  // When true, any unsupported options will be ignored and OK will be returned
+  bool ignore_unsupported_options = true;
```

Comment:
What' the difference between `ignore_unknown_options` and `ignore_unsupported_options`? Are the "unsupported" options a subset of "known options"？

Reply:
Unsupported options are the ones who, when being configured, return Status::NotSupported.  These are potentially valid options for the object under certain conditions but fail in this case.  For example, the CuckooTable might be a valid "table_factory" but was not enabled or compiled in.  In this case, requesting to configure a table_factory=CuckooTable would return NotSupported.  Unsupported implies that, the named option is a valid option but for some reason the value is not valid under these circumstances.

Unknown options are a catch-all for a lot of errors, and are already used as such.  An Unknown option might be one were the name is not valid (e.g. "no_such_option=value") or it might be one who has an illegal value to int (my_int=fred).